app/js/script.js

```diff
@@ -0,0 +1,160 @@
+let dragging = false;
+let clockInterval;
+let selectedTimeZone = 'UTC';
+let lastTime = { hours: 0, minutes: 0, seconds: 0 };
+
+const hourHand = document.querySelector('#hour');
+const minuteHand = document.querySelector('#minute');
+const secondHand = document.querySelector('#second');
+
+const timeZones = [
+  'UTC',
+  'America/New_York',
+  'Europe/London',
+  'Asia/Tokyo',
+  'Australia/Sydney'
+];
+
+function startClock() {
+  addTimeZoneSelector();
+  initializeTime();
+
```

Comment:
Functions addTimeZoneSelector and initializeTime above are called before they are declared. Advised moving the declarations above the calls